src/sqlfluff/dialects/dialect_spark3.py

```diff
@@ -1515,6 +1515,28 @@ class SamplingExpressionSegment(BaseSegment):
     )
 
 
+@spark3_dialect.segment()
+class LateralViewClauseSegment(BaseSegment):
+    """A `LATERAL VIEW` like in a `FROM` clause.
+
+    https://spark.apache.org/docs/latest/sql-ref-syntax-qry-select-lateral-view.html
+    """
+
+    type = "lateral_view_clause"
+
+    match_grammar = Sequence(
+        Indent,
+        "LATERAL",
+        "VIEW",
+        Ref.keyword("OUTER", optional=True),
+        Ref("FunctionSegment"),
+        AnyNumberOf(
+            Ref("AliasExpressionSegment"),
+        ),
```

Comment:
One last question, I promise!

Wouldn't this allow this?:

```sql
    LATERAL VIEW OUTER EXPLODE(ARRAY()) AS tbl_name AS c_age;
```

Reply:
Please keep asking 🙂  Updated (and will include CI updates as well) to fix this.

Using the below instead.

```python
        Ref("SingleIdentifierGrammar", optional=True),
        Ref("AliasExpressionSegment", optional=True),
```

I ran the query with `AS tbl_name AS c_age;` and it failed. The first `AS` isn't allowed like you raised.

Reply:
Tested the second `AS` and going to push one more commit then should be good to go

Reply:
Also removed two other test cases that were not valid so thank you for catching that initial case.